lua/barbar.lua

```diff
@@ -6,13 +6,13 @@ local command = vim.api.nvim_command --- @type function
 local create_user_command = vim.api.nvim_create_user_command --- @type function
 local set_option = vim.api.nvim_set_option --- @type function
 
-local api = require'barbar.api'
-local bbye = require'barbar.bbye'
-local events = require'barbar.events'
-local notify = require'barbar.utils'.notify
-local render = require'barbar.render'
-local state = require'barbar.state'
-local utils = require'barbar.utils'
+local Api = require('barbar.api')
+local Bbye = require('barbar.bbye')
+local Events = require('barbar.events')
```

Comment:
I'm not convinced about using PascalCase for our imports.

Reply:
The reason I ended up wanting to do this was, while working on the previous PR, I noticed that we had something like this:

```lua
--- @class barbar.nodes: barbar.node[]
local m

--- @param bar barbar.nodes
function m.foo(bar)
end
```

Autocompletion started suggesting `bar.foo()` as a valid function call, but we were just passing in a list (e.g. `nodes.foo({…})`). I had mistakenly annotated that a class instance was going to be passed in; renaming `nodes` to `Nodes` made that clear to me. I wanted to see if that mistake had been made anywhere else, but the clearest way to do it was to change the case in order to differenciate. I figured I would leave it to prevent it from happening again

Reply:
> Are we sure @class is the correct annotation for modules?

Yep. I just checked some of folke's code, and the only difference is that he defines all of the `@class`es in a separate file and then does `@type` on the module (whereas `@class` tells the language server the type when we do it). Functionally the same, I just prefer putting the annotations in the file where they're defined.

> And isn't `local m` idiomatic for lua?

Yes, though that wasn't the point of the example.

The important part was:

```lua
--- @class barbar.nodes: barbar.node[] ← this name
local could_be_anything

--- @param bar barbar.nodes ← matches this name
function could_be_anything.foo(bar)
end
```

And thus the LSP wrongly thinks `bar` can use `could_be_anything`'s functions (i.e. bar.foo()`).

> And I'd rather we make the existing Modules into modules than the other way around.

So how can we prevent it from happening by accident in the future? A naming convention seems like the best way to quickly identify which types are data and which ones the LSP thinks are instances.